pkg/featuregate/featuregate_test.go

```diff
@@ -0,0 +1,407 @@
+/*
```

Comment:
I would assume that this featuregate code lives elsewhere, primarily. Do we need those tests?

Reply:
I figured these tests were nice in case we really did something stupid and / or changed the underlying implementation some day. I'm fine with removing them in favor of just having tests in pkg/flags instead though.